### ISSUE TITLE
Don't vendor Ranged-sets?

### ISSUE THREAD

**Issue by Ericson2314** [simonmar/alex#182]
Right now, https://hackage.haskell.org/package/Ranged-sets depends on big deps like HUnit and QuickCheck. But I have a PR https://hackage.haskell.org/package/Ranged-sets so there is a flag to disable the arbitrary instances and it just depends on `base`. At that point, I don't see why we just delete the vendored code and use that instead?

**Comment by andreasabel** [comment]
A reason could be that atm, Alex depends only on libraries that ship with GHC (https://wiki.haskell.org/Core_Libraries_Committee#Core_Libraries).  This _might_ be a property worth preserving.  (But I don't have insight into the GHC build process, for instance.)

**Comment by Ericson2314** [comment]
Yeah I won't pull the trigger on this until I bring up on ghc-devs. Maybe we could have a flag to either use or not use the vendored part as a stop-gap to track divergence too.
